src/FSharp.Data.GraphQL.Server.AspNetCore/Giraffe/HttpHandlers.fs

```diff
@@ -1,155 +1,268 @@
+                                logger.LogTrace($"GraphQL deferred errors:{Environment.NewLine}{{errors}}", errors)
+                        | DeferredErrors(data, errors, path) ->
+                            logger.LogInformation(
+                                "Produced GraphQL deferred result with errors for path: {path}",
+                                path |> Seq.map string |> Seq.toArray |> Path.Join
+                            )
+
+                            if logger.IsEnabled LogLevel.Trace then
+                                logger.LogTrace(
+                                    $"GraphQL deferred errors:{Environment.NewLine}{{errors}}{Environment.NewLine}GraphQL deferred data:{Environment.NewLine}{{data}}",
+                                    errors,
+                                    serializeIdented data
+                                ))
+
+                GQLResponse.Direct(documentId, data, errs)
+            | Stream stream ->
+                logger.LogInformation(
+                    $"Produced stream GraphQL response with documentId = '{{documentId}}' and metadata:{Environment.NewLine}{{metadata}}",
+                    documentId,
+                    metadata
+                )
+
+                if logger.IsEnabled LogLevel.Information then
+                    stream
+                    |> Observable.add (function
+                        | SubscriptionResult data ->
+                            logger.LogInformation("Produced GraphQL subscription result")
+
+                            if logger.IsEnabled LogLevel.Trace then
+                                logger.LogTrace(
+                                    $"GraphQL subscription data:{Environment.NewLine}{{data}}",
+                                    serializeIdented data
+                                )
+                        | SubscriptionErrors(null, errors) ->
+                            logger.LogInformation("Produced GraphQL subscription errors")
+
+                            if logger.IsEnabled LogLevel.Trace then
+                                logger.LogTrace($"GraphQL subscription errors:{Environment.NewLine}{{errors}}", errors)
+                        | SubscriptionErrors(data, errors) ->
+                            logger.LogInformation("Produced GraphQL subscription result with errors")
+
+                            if logger.IsEnabled LogLevel.Trace then
+                                logger.LogTrace(
+                                    $"GraphQL subscription errors:{Environment.NewLine}{{errors}}{Environment.NewLine}GraphQL deferred data:{Environment.NewLine}{{data}}",
+                                    errors,
+                                    serializeIdented data
+                                ))
+
+                GQLResponse.Stream documentId
+
+        /// Checks if the request contains a body
+        let checkIfHasBody (request: HttpRequest) = task {
```

Comment:
not a big deal: maybe checking the content-length header first (not only, but first) is slightly more performant for most situations.